test/unit/userTest.ts

```diff
@@ -127,6 +127,36 @@ describe('User Tests', () => {
         });
       }
     });
+
+    it('Should generate length 12 and complexity 5 password containing at least one number and one  lowercase letter ', () => {
+      const passwordCondition = { length: 12, complexity: 5 };
+      const password: SecureBuffer<void> = User.generatePasswordUtf8(passwordCondition);
+      password.value((buffer: Buffer): void => {
+        const passwordAsArrayOfCharacters = buffer.toString('utf8');
+        const complexity5Regex = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$|%^&*()[]_-])(?=.{12})');
+
+        expect(complexity5Regex.test(passwordAsArrayOfCharacters));
+      });
+    });
+    it('Should generate length 14 and complexity 3 password containing at least one number and one  lowercase letter ', () => {
+      const passwordCondition = { length: 14, complexity: 3 };
+      const password: SecureBuffer<void> = User.generatePasswordUtf8(passwordCondition);
+      password.value((buffer: Buffer): void => {
+        const passwordAsArrayOfCharacters = buffer.toString('utf8');
+        const complexity3Regex = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{14})');
+
+        expect(complexity3Regex.test(passwordAsArrayOfCharacters));
+      });
+    });
+
+    it('Should throw an error beacuse of complexity', () => {
+      try {
+        const passwordCondition = { length: 14, complexity: 9 };
+        User.generatePasswordUtf8(passwordCondition);
```

Comment:
you need to add something like `assert.fail('the above should throw an error')` below `generatePasswordUtf8()` so incase the function doesn't throw an error, the test will fail